docs/routing/introduction.md

```diff
@@ -147,6 +147,38 @@ To access the [`router` object](/docs/api-reference/next/router.md#router-object
 
 In general we recommend using [`useRouter`](/docs/api-reference/next/router.md#useRouter).
 
+## Prefetching pages
+
+Next.js can prefetch pages for fast transitions even when networks are slow.
+
+Any `<Link />` in the viewport (initially or through scroll) will be automatically prefetched by default. Pages using [Static Generation](/docs/basic-features/data-fetching/get-static-props.md) will preload `JSON` files with the data for faster page transitions.
+
+Next.js also allows you to prefetch server-rendered pages using HTML `<link rel="prefetch"/>` tags:
+
+> [Link prefetching](https://developer.mozilla.org/en-US/docs/Web/HTTP/Link_prefetching_FAQ) is a browser mechanism, which utilizes browser idle time to download or prefetch documents that the user might visit in the near future. A web page provides a set of prefetching hints to the browser, and after the browser is finished loading the page, it begins silently prefetching specified documents and stores them in its cache. When the user visits one of the prefetched documents, it can be served up quickly out of the browser's cache.
+
+```javascript
+import Link from 'next/link'
+import Head from 'next/head'
+
+// Prefetch the JSON data that will be loaded on the server
+// for `getServerSideProps`. Note that this is a separate route
+// and replies on Next.js internals (_next/data) and is subject
+// to change in the future. Use this example with caution.
+export default function Home() {
+  return (
+    <>
+      <Head>
+        <link rel="prefetch" href="/_next/data/buildId/about.json">
```

Comment:
This can be a bit trickier due to the encoding needing to match and all query parameters must match as well. 

Maybe we should add a flag to the router to allow adding this on the user's behalf 🤔

Reply:
Yeah that PR needs a bit more work, no rush here 👍